Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,332 @@
+        AZStd::string root_name(m_model->root_link_->name.c_str(), m_model->root_link_->name.size());
+        created_links[root_name] = createEntityRoot;
+        if (!createEntityRoot.IsSuccess())
+        {
+            return AZ::Failure(AZStd::string(createEntityRoot.GetError()));
+        }
+
+        auto links = Utils::GetAllLinks(m_model->root_link_->child_links);
+
+        // create links
+        for (const auto& [name, link_ptr] : links)
+        {
+            created_links[name] = AddEntitiesForLink(link_ptr, createEntityRoot.GetValue());
+        }
+
+        for (const auto& [name, result] : created_links)
+        {
+            AZ_TracePrintf(
+                "CreatePrefabFromURDF",
+                "Link with name %s was created as: %s",
+                name.c_str(),
+                result.IsSuccess() ? (result.GetValue().ToString().c_str()) : ("[Failed]"));
+            AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+            if (result.IsSuccess())
+            {
+                m_status.emplace(name, AZStd::string::format("created as: %s", result.GetValue().ToString().c_str()));
+            }
+            else
+            {
+                m_status.emplace(name, AZStd::string::format("failed : %s", result.GetError().c_str()));
+            }
+        }
+
+        // set transforms of links
+        for (const auto& [name, link_ptr] : links)
+        {
+            const auto this_entry = created_links.at(name);
+            if (this_entry.IsSuccess())
+            {
+                AZ::Transform tf = Utils::GetWorldTransformURDF(link_ptr);
+                auto* entity = AzToolsFramework::GetEntityById(this_entry.GetValue());
+                if (entity)
+                {
+                    auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+                    if (transformInterface)
+                    {
+                        AZ_TracePrintf(
+                            "CreatePrefabFromURDF",
+                            "Setting transform %s %s to [%f %f %f] [%f %f %f %f]",
+                            name.c_str(),
+                            this_entry.GetValue().ToString().c_str(),
+                            tf.GetTranslation().GetX(),
+                            tf.GetTranslation().GetY(),
+                            tf.GetTranslation().GetZ(),
+                            tf.GetRotation().GetX(),
+                            tf.GetRotation().GetY(),
+                            tf.GetRotation().GetZ(),
+                            tf.GetRotation().GetW());
+                        transformInterface->SetWorldTM(tf);
+                    }
+                    else
+                    {
+                        AZ_TracePrintf(
+                            "CreatePrefabFromURDF", "Setting transform failed: %s does not have transform interface", name.c_str());
```

Comment:
This `AZ_TracePrintf` should have a newline at the end
```suggestion
                        AZ_TracePrintf(
                            "CreatePrefabFromURDF", "Setting transform failed: %s does not have transform interface\n", name.c_str());
```